### PR TITLE
Issue#20915: check for undefined index

### PR DESCRIPTION
**Pull request by vinayshah**
### Fixed Issues (if relevant)

1. magento/magento2#20915: Import Product through CSV from extended module, face un index issue from Abstract Class

### Manual testing scenarios (*)

Facing specific through our extension, though it's a general fix.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=20916) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @vinayshah. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by sivaschenko** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, here is your [Magento instance](https://i-20916-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-20916-2-3-develop.instances.magento-community.engineering/admin](https://i-20916-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-4098](https://jira.corp.magento.com/browse/ENGCOM-4098) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
Hi @vinayshah I was able to reproduce this issue only with code manipulation. Could you provide csv file  example that caused such error to verify PR changes?

**Comment by sidolov** [comment]
@vinayshah , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by vinayshah** [comment]
Hello @sidolov @VasylShvorak, the issue has occurred more of due too specific to our extension, The default Magento flow doesn't allow to add a new attribute value, it only adds existing value. To add new value it creates issue. For this, we need to overwrite the complete code instead of simple isset statement.

File: app/code/Magento/CatalogImportExport/Model/Import/Product/Type/AbstractType.php [hunk 1]
```diff
@@ -5,16 +5,14 @@
...
-use Magento\CatalogImportExport\Model\Import\Product\RowValidatorInterface;
 use Magento\CatalogImportExport\Model\Import\Product;
+use Magento\CatalogImportExport\Model\Import\Product\RowValidatorInterface;
+use Magento\Framework\App\ResourceConnection;
 use Magento\Framework\EntityManager\MetadataPool;
 
 /**
  * Import entity abstract product type model
  *
- * @api
```
Comment (sidolov) [inline app/code/Magento/CatalogImportExport/Model/Import/Product/Type/AbstractType.php:16]: Please, revert annotation removal

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
@vinayshah  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**